aws/pg/placement_group.go

```diff
@@ -0,0 +1,103 @@
+package pg
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/coinbase/odin/aws"
+	"github.com/coinbase/step/utils/to"
+)
+
+// FindOrCreatePartitionGroup will find a partition group by name.
+// If it doenst exist:
+// - It will create one with the provided detail (error if detail is nil)
+// If one does exist:
+// - It will fetch that Partition Group and Error if any provided detail is missing
+func FindOrCreatePartitionGroup(ec2c aws.EC2API, prefix string, groupName *string, partitionCount *int64, strategy *string) error {
+	if groupName == nil {
+		return fmt.Errorf("PlacementGroupError: groupName nil")
+	}
+
+	pg, err := findPlacementGroup(ec2c, groupName)
+	if err != nil {
+		return err
+	}
+
+	// If no pg is found create a new one
+	// otherwise validate that the found one equals the correct values
+	if pg == nil {
+		// If we are creating the Placement Group we restrict the name to prefix
+		if !strings.HasPrefix(*groupName, prefix) {
```

Comment:
Would be nice if validation was outside here

Reply:
We want to allow projects to use other projects placement groups, I just want to ensure that placement groups created by odin have the correct prefix. 

This is not ideal, but doing any sort of auto-name assignment makes this harder to do.